packages/labs/router/README.md

```diff
@@ -4,4 +4,193 @@ A router for Lit.
+  private routes = new Routes(this, [
+    {path: '/', render: () => html`<h1>Home</h1>`},
+    {path: '/projects', render: () => html`<h1>Projects</h1>`},
+    {path: '/about', render: () => html`<h1>About</h3>`},
+  ]);
+}
+```
+
+The second argument is the route configuration: an array of `RouteConfig` objects.
+
+#### RouteConfig
+
+A RouteConfig contains at the minimum the pattern to match URLs against and a template to render. Names can be provided to reference routes for link generation.
+
+There are two types of RoutConfig: `PathRouteConfig` and `URLPatternRouteConfig`:
+
+`PathRouteConfig` lets you specify the URL pattern as a path string:
+
+```ts
+{name: 'home', path: '/', render: () => html`<h1>Home</h1>`}
+```
+
+```ts
+export interface PathRouteConfig {
+  name?: string | undefined;
+  path: string;
+  render: (params: {[key: string]: string}) => unknown;
+}
+```
+
+`URLPatternRouteConfig` lets you specify the URL pattern as a [`URLPattern`](https://developer.mozilla.org/en-US/docs/Web/API/URLPattern) object:
+
+```ts
+{pattern: new URLPattern({pathname: '/'}), render: () => html`<h1>Home</h1>`}
+```
+
+```ts
+export interface URLPatternRouteConfig {
+  name?: string | undefined;
+  pattern: URLPattern;
+  render: (params: {[key: string]: string}) => unknown;
+}
+```
+
+#### Render callbacks
+
+The render callback is called when the outlet property of the Routes object is accessed. It is passed an object with the parameters extracted from the matching URL.
+
+Example with named parameter:
+
+```ts
+{
+  path: '/profile/:id',
+  render: ({id}) => html`<x-profile .profileId=${id}></x-profile>`
+}
+```
+
+#### Outlets
+
+An outlet is where a routes object renders the currently selected route's template. It can be used anywhere in the host element's template:
+
+```ts
+html`${this.routes.outlet}`;
+```
+
+#### `goto()`
+
+`goto(url: string)` is a programmatic navigation API. It should be able to take full URLs for top-level navigation and relative URLs for navigation within a nested route space.
+
+`goto(name: string, params: object)` _(not implemented)_ allows navigation via named routes. The name and params are scoped to the Routes object it's called on, though nested routes can be triggered by a "tail" parameter - the match of a trailing /\*parameter (See tail groups).
+
+#### `link()`
+
+Components need to generate links to resources within the app. It's desirable to not require that link generation has global knowledge of URLs, so links should be able to be generated with either only local information, or with abstracted parameters. `routes.link()` helps generate different kinds of links:
+
+##### Relative links
+
+Relative links are relative to the parent route and its current state, and can be specified with a path string or name and parameters.
+
+Examples
+
+- `this._routes.link(user.id)` - within a `<x-user>` component that has a route pattern like ':id', this would link to another user profile.
```

Comment:
```suggestion
- `this._routes.link(user.id)` - within a `<x-user>` component that has a route pattern like `":id"`, this would link to another user profile.
```